kiegroup/kogito-editors-java
# KOGITO-5225/KOGITO-5154: Support React components on Kogito Java-editors

### JIRAs

- [KOGITO-5225](https://issues.redhat.com/browse/KOGITO-5225): New boxed expression editor - React Components module pipeline

- [KOGITO-5154](https://issues.redhat.com/browse/KOGITO-5154): Import Java classes as data objects on Kogito - Integrate with GWT

### Description

This PR introduces the infrastructure for supporting React components on Kogito editors (DMN and BPMN). They live in the `packages` directory and follow the convention of **components** and **loaders**:
- **Components** - all elements in the editors (a marshaller, a dialog, or even a FEEL parser)
- **Loaders** - enable editors to load and render their components

The included packages are:
- **`packages/hello-world-component`** - it's a simple component used for testing purposes (my following PR will replace it with the `boxed-expression-editor`)
- **`packages/dmn-loader`** - responsible for loading components in the DMN editor

### How to test this PR

You may render the **Hello world** component to test this PR locally. Add the following lines the `DMNClientEntryPoint.java`, and run the `kie-wb-common-dmn-webapp-kogito-testing` to see the React component rendering in the GWT world:

```java
final Element container = DomGlobal.document.createElement("div");

container.id = "container";
DomGlobal.document.body.appendChild(container);

DMNLoader.renderFeelDialog("#container");
```

Alternatively, you may run the `kie-wb-common-dmn-webapp-kogito-testing` and paste this on browser console:

```javascript
const container = document.createElement("div");
container.id = "container";
document.body.appendChild(container);
__KIE__DMN_LOADER__.renderHelloWorld("#container");
```

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>
